File: frontend/test_bedrock.js
const fs = require('fs');
const path = require('path');
const { BedrockRuntimeClient, InvokeModelCommand } = require('@aws-sdk/client-bedrock-runtime');

async function test() {
  const envPath = path.join(process.cwd(), '../.env');
  const envContent = fs.readFileSync(envPath, 'utf8');
  const matchAccess = envContent.match(/^2_ACCESS_KEY_ID=(.*)$/m);
  const matchSecret = envContent.match(/^2_SECRET_ACCESS_KEY=(.*)$/m);
  const matchRegion = envContent.match(/^2_REGION=(.*)$/m);
  
  const client = new BedrockRuntimeClient({
    region: matchRegion[1].trim(),
    credentials: {
      accessKeyId: matchAccess[1].trim(),
      secretAccessKey: matchSecret[1].trim()
    }
  });

  try {
    const command = new InvokeModelCommand({
      modelId: 'amazon.titan-embed-text-v2:0',
      contentType: 'application/json',
      accept: 'application/json',
      body: JSON.stringify({ inputText: 'test', dimensions: 1024, normalize: true })
    });
    const res = await client.send(command);
    console.log("Success");
  } catch (err) {
    console.error("AWS Error:", err);
  }
}
test();

File: scratch/test_bedrock.py
import boto3
import os
from dotenv import load_dotenv

load_dotenv()

client = boto3.client(
    "bedrock-runtime",
    region_name=os.getenv("BEDROCK_REGION", "ap-southeast-1"),
    aws_access_key_id=os.getenv("2_ACCESS_KEY_ID"),
    aws_secret_access_key=os.getenv("2_SECRET_ACCESS_KEY")
)

try:
    response = client.invoke_model(
        modelId="amazon.titan-embed-text-v1",
        body=b'{"inputText": "Hello world"}'
    )
    print("V1 success")
except Exception as e:
    print(f"V1 error: {e}")

try:
    response = client.invoke_model(
        modelId="amazon.titan-embed-text-v2:0",
        body=b'{"inputText": "Hello world"}'
    )
    print("V2 success")
except Exception as e:
    print(f"V2 error: {e}")
